material/notebooks/Download.ipynb

```diff
@@ -0,0 +1,74 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Downloading images\n",
+    "\n",
+    "In this notebook I will show how to use the `download` module found at `../data/download.py`\n",
+    "We will use it to download labeled and unlabeled images (train and test sets respectively).\n",
+    "\n",
+    "The script can scale well to the huge amount of images we deal with it because:\n",
+    "\n",
+    "1. It can be paused and restarted, in which case it skips the already existing images\n",
+    "2. It uses multiprocessing to speed-up the process\n",
+    "3. It reports the progress using a progress bar.\n",
+    "\n",
+    "**Let's see it in action!**"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import sys\n",
+    "sys.path.append(\"../data\") # Append source directory to our Python path\n",
+    "\n",
+    "from download import dl_images"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Let's download some labeled images.\n",
+    "train_file = \"../data/train.json\"\n",
+    "train_dir = \"../data/train/\"\n",
+    "\n",
+    "dl_images(data_file, out_dir, processes=20)\n",
```

Comment:
It is not clear from the notebook what the values of 'data_file' and 'out_dir' should be. Further explanation would be quite helpful for the user.

Reply:
I fixed and pushed a commit to correct a typo. Regarding the explanation of the parameters, the user is supposed to check the docstring of the function.

```
def dl_images(data_file, out_dir, processes=20):
    """Download all images found in the metadata file to the specified folder.

    Parameters
    ----------
    data_file : str
        Path to the JSON file containing the image metadata (id, url and potentially label)
    out_dir : str
        Path to the folder where the images will be saved.
    processes : int, optional
        Number of OS processes to be used. Increasing this number may speed-up the process but will also
        use more RAM and CPU.

    """
```

I think the arguments are then quite clear, right?